modules/eks/actions-runner-controller/charts/actions-runner/templates/runnerdeployment.yaml

```diff
@@ -43,5 +43,27 @@ spec:
       imagePullPolicy: IfNotPresent
       serviceAccountName: {{ .Values.service_account_name }}
       resources:
-        {{- toYaml .Values.resources | nindent 8 }}
+        limits:
+          cpu: {{ .Values.resources.limits.cpu }}
+          memory: {{ .Values.resources.limits.memory }}
+          {{- if and .Values.dind_enabled .Values.resources.limits.ephemeral_storage }}
+          ephemeral-storage: {{ .Values.resources.limits.ephemeral_storage }}
+          {{- end }}
+        requests:
+          cpu: {{ .Values.resources.requests.cpu }}
+          memory: {{ .Values.resources.requests.memory }}
+      {{- if and .Values.dind_enabled .Values.storage }}
+      dockerVolumeMounts:
+      - mountPath: /var/lib/docker
+        name: docker-volume
+      volumes:
+      - name: docker-volume
+        ephemeral:
+          volumeClaimTemplate:
+            spec:
+              accessModes: [ "ReadWriteOnce" ] # Only 1 pod can connect at a time
+              resources:
+                requests:
+                  storage: {{ .Values.storage }}
+      {{- end }}
```

Comment:
Can't we just use the host filesystem? Do we really need to allocate an additional volume for ephemeral storage?

Reply:
Because the customer app image we are building exceeds 20Gb. With this additional volume for ephemeral storage, we can attach a temporary block storage while building and testing the app with the given runner, rather than increasing the storage across the entire host filesystem